ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/NoUnusedImportsRule.kt

```diff
@@ -112,5 +126,43 @@ class NoUnusedImportsRule : Rule("no-unused-imports") {
         }
     }
 
+    // Builds a list of imports. Takes care of having aliases in case it is assigned to imports
+    private fun buildNonRedundantImports(text: String) {
+        val element = text.split("import").last().trim()
+        if (element.contains("as ")) {
+            imports.addAll(listOf(element.split("as").first().replace("`", "").trim(), element.split("as").last().trim()))
```

Comment:
```suggestion
            imports.addAll(listOf(element.split(" as ").first().replace("`", "").trim(), element.split(" as ").last().trim()))
```

There is a bug here when the import contains the letter "as" for example `com.fastcompany.foo.Anything` this will cut the whole import in the wrong place and end up reporting unused import

Reply:
If you use the following test is going to fail
```
    @Test
    fun `Should not report unused here`() {
        assertThat(
            NoUnusedImportsRule().lint(
                """
                package com.fastcompany.foo

                import com.fastcompany.common.Awesome as CommonAsset

                class SomeConfig {
                
                    fun fooFunction() = CommonAsset()
                }

                """.trimIndent()
            )
        ).isEmpty()
    }
```

Reply:
Thank you for this test ! In addition to it, i changed the contains condition above it also to `contains(" as ") ` and added one more test to check that.